tools/hack/create-cluster.sh

```diff
@@ -53,10 +55,31 @@ kubectl rollout status -n metallb-system deployment/controller --timeout 5m
 kubectl rollout status -n metallb-system daemonset/speaker --timeout 5m
 
 # Apply config with addresses based on docker network IPAM.
-subnet=$(docker network inspect kind | jq -r '.[].IPAM.Config[].Subnet | select(contains(":") | not)')
-# Assume default kind network subnet prefix of 16, and choose addresses in that range.
-address_first_octets=$(echo "${subnet}" | awk -F. '{printf "%s.%s",$1,$2}')
-address_range="${address_first_octets}.255.200-${address_first_octets}.255.250"
+address_ranges=""
+
+if [ "${IP_FAMILY}" = "ipv4" ] || [ "${IP_FAMILY}" = "dual" ]; then
+    subnet_v4=$(docker network inspect kind | jq -r '.[].IPAM.Config[] | select(.Subnet | contains(":") | not) | .Subnet')
+    address_prefix_v4=$(echo "${subnet_v4}" | awk -F. '{print $1"."$2"."$3}')
+    address_range_v4="${address_prefix_v4}.200-${address_prefix_v4}.250"
+    echo "IPv4 address range: ${address_range_v4}"
+    address_ranges+="- ${address_range_v4}"
+fi
+
+if [ "${IP_FAMILY}" = "ipv6" ] || [ "${IP_FAMILY}" = "dual" ]; then
+    subnet_v6=$(docker network inspect kind | jq -r '.[].IPAM.Config[] | select(.Subnet | contains(":")) | .Subnet')
+    ipv6_prefix="${subnet_v6%::*}"
+    address_range_v6="${ipv6_prefix}::200-${ipv6_prefix}::250"
```

Comment:
Nit: is this decimal or hex?

Reply:
Oh I have now explicitly changed it to hex.